pkg/ccl/backupccl/restore_online.go

```diff
@@ -392,21 +411,24 @@ func (r *restoreResumer) waitForDownloadToComplete(
 			}
 			lastProgressUpdate = timeutil.Now()
 		}
+		// Signal the download job if it is waiting that we've polled and found work
+		// left for it to do.
+		select {
+		case ch <- struct{}{}:
```

Comment:
it would be nice to have a unit test for this change, but it probably requires testing knobs. You could add something like: assert that the number of empty structs sent equals the number re-attempts by the download span worker. This feels less heavy than trying to simulate the race, but also checks that this logic does work correctly with and without the race.

Reply:
> requires testing knobs

Yeah, I think other than actually getting the race in pebble, what we'd do is add two knobs:
on to _block_ the poller loop, then another after the download rpc, so that we do a download pass, then go insert a new external file that it "missed" (since we added it after, then unblock the poller loop so it observes the incomplete download. that would hang without the addition of the loop around download.

I thought about this (and discussed it on slack) but it doesn't seem worth it. Unskipping the existing test and having it not fail (I ran it 2000 times) also serves to validate the new behavior, without a bunch of new knobs.

Reply:
I suppose, I think the only thing my proposal would add is an assertion that we don't inadvertently loop through download spans unecessarily. but i'm fine merging for now. we've got bigger fish to fry.

Reply:
It's pretty much a noop to loop through it again. I'm inclined to just move on.